File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int el = 0;
        int cnt = 0;
        for(int i=0;i<nums.length;i++){
            if(cnt==0){
                el= nums[i];
                cnt = 1;
            }
            else if(el==nums[i]){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        return el;
    }
}

File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 1;
        int el = nums[0];
        for(int i=1;i<nums.size();i++){
            if(nums[i]==el){
                count++;
            }
            else{
                count--;
                if(count==0){
                    el = nums[i];
                    count=1;
                }
            }
        }
        return el;
    }
};